src/python/gem5/simulate/exit_event_generators.py

```diff
@@ -28,6 +28,9 @@
 from ..components.processors.abstract_processor import AbstractProcessor
 from ..components.processors.switchable_processor import SwitchableProcessor
 from m5.util import warn
+from os.path import join as joinpath
+from pathlib import Path
```

Comment:
Do we need both of these?

If using pathlib, I think you can use this instead: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.joinpath